### ISSUE TITLE
split react and non-react rules

### ISSUE THREAD

**Issue by itajaja** [rackt/eslint-config-rackt#3]
If a package doesn't depend on react (e.g. history, redux), it should be nice to still use eslint-config-rackt without the react rules. There are two approaches I can see:

````js
// for repos with react dependency
"extends": "rackt", "rackt-react"

// without react dependency
"extends": "rackt"
````
This one will require the existing repos to change their eslintrc file

The other approach would be

````js
// for repos with react dependency
"extends": "rackt"

// without react dependency
"extends": "rackt-no-react"
````

This will allow the existing repos to remain unchanged. Not sure if this involves creating two separate repos...

cc @taion

**Comment by taion** [comment]
IMO the best way to handle this is to do something like: https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb

Where we'd have e.g. `rackt` and `rackt/base` (probably no `rackt/legacy`).

**Comment by gaearon** [comment]
:+1: to @taion's suggestion

**Comment by taion** [comment]
Really what I'd like to do what be to split the rules up into sections, the way it's done on eslint-config-defaults and eslint-config-airbnb.

But frankly after having done that once for eslint-config-airbnb, this kind of thing is well into the "never again" territory for me, so, uhh, I'm not going to do that.

**Comment by taion** [comment]
Wait, @itajaja, there are no React rules here as far as I can tell. The issue should be more like bringing in the React rules for something like `rackt/react`: https://github.com/rackt/react-router/blob/master/.eslintrc#L3-L10

**Comment by itajaja** [comment]
You are right, I am probably talking about jsx rules ("jsx-quotes") but probably that's not a problem to have them around...

**Comment by taion** [comment]
Nope, they're fine - and they don't require the React plugin either. I'm going to edit this issue title to reflect what's actually missing.

**Comment by taion** [comment]
Although given that Rackt are supposed to be "Quality code from the React.js community", maybe `rackt` and `rackt/base` makes more sense.

Then again, given Redux and history, the proportion of React projects in Rackt seems to on a downhill trajectory :stuck_out_tongue_closed_eyes:
